src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -159,5 +198,164 @@ public void HotReloadConfigRestRuntimeOptions()
+            fileSystem.File.WriteAllText(configName, initialConfig.ToJson());
+            FileSystemRuntimeConfigLoader configLoader = new(fileSystem, handler: null, configName, string.Empty);
+            RuntimeConfigProvider configProvider = new(configLoader);
+            RuntimeConfig lkgRuntimeConfig = configProvider.GetConfig();
+
+            Assert.IsNotNull(lkgRuntimeConfig);
+
+            //Act
+            // Simulate a wrong change to the schema file and updating config correctly
+            fileSystem.File.WriteAllText(schemaName, schemaConfig);
+            fileSystem.File.WriteAllText(configName, updatedConfig.ToJson());
+
+            // Give ConfigFileWatcher enough time to hot reload the change
+            System.Threading.Thread.Sleep(1000);
+
+            try
+            {
+                configProvider.GetConfig();
+            }
+            catch (DataApiBuilderException dbException)
+            {
+                Assert.AreEqual(expected: "Failed validation of configuration file.", actual: dbException.Message);
+            }
+
+            RuntimeConfig newRuntimeConfig = configProvider.GetConfig();
+            Assert.AreEqual(expected: lkgRuntimeConfig, actual: newRuntimeConfig);
+
+            if (File.Exists(configName))
+            {
+                File.Delete(configName);
+            }
+
+            if (File.Exists(schemaName))
+            {
+                File.Delete(schemaName);
+            }
+        }
+
+        /// <summary>
+        /// Creates a hot reload scenario in which the updated configuration file is wrong so
+        /// hot reload to fail, then we check that the program is still able to work
+        /// properly by showing us that it is still using the same configuration file
+        /// from before the hot reload.
+        /// </summary>
+        [TestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        public void HotReloadParsingFail()
+        {
+            //Arrange
+            string schemaName = "dab.draft.schema.json";
+            string configName = "hotreload-config.json";
+            if (File.Exists(configName))
+            {
+                File.Delete(configName);
+            }
+
+            bool initialRestEnabled = true;
+
+            bool initialGQLEnabled = true;
+
+            DataSource dataSource = new(DatabaseType.MSSQL,
+                ConfigurationTests.GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL), Options: null);
+
+            RuntimeConfig initialConfig = new(
+                Schema: schemaName,
+                DataSource: dataSource,
+                Runtime: new(
+                    Rest: new(initialRestEnabled),
+                    GraphQL: new(initialGQLEnabled),
+                    Host: new(null, null, HostMode.Development)
+                ),
+                Entities: new(new Dictionary<string, Entity>())
+            );
+
+            string updatedConfig = GenerateWrongRuntimeSection();
```

Comment:
The previous test and this differ only in the fact that they generate a different wrong section. Can we somehow have a common function to minimize code duplication?